runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/PubsubIOTranslator.java

```diff
@@ -71,38 +66,33 @@ public void translate(
   /**
    * Implements PubsubIO Write translation for the Dataflow backend.
    */
-  public static class WriteTranslator<T>
-      implements TransformTranslator<DataflowPipelineRunner.StreamingPubsubIOWrite<T>> {
+  public static class WriteTranslator<T> implements TransformTranslator<PubsubUnboundedSink<T>> {
 
     @Override
     @SuppressWarnings({"rawtypes", "unchecked"})
     public void translate(
-        DataflowPipelineRunner.StreamingPubsubIOWrite transform,
+        PubsubUnboundedSink<T> transform,
         TranslationContext context) {
       translateWriteHelper(transform, context);
     }
 
     private <T> void translateWriteHelper(
-        DataflowPipelineRunner.StreamingPubsubIOWrite<T> customTransform,
+        PubsubUnboundedSink<T> transform,
         TranslationContext context) {
-      if (!context.getPipelineOptions().isStreaming()) {
-        throw new IllegalArgumentException(
-            "PubsubIO.Write is non-primitive for the Dataflow batch runner.");
-      }
-
-      PubsubIO.Write.Bound<T> transform = customTransform.getOverriddenTransform();
+      Preconditions.checkState(context.getPipelineOptions().isStreaming(),
+          "PubsubIOTranslator.WriteTranslator is only for streaming pipelines.");
 
-      context.addStep(customTransform, "ParallelWrite");
+      context.addStep(transform, "ParallelWrite");
       context.addInput(PropertyNames.FORMAT, "pubsub");
-      context.addInput(PropertyNames.PUBSUB_TOPIC, transform.getTopic().asV1Beta1Path());
+      context.addInput(PropertyNames.PUBSUB_TOPIC, transform.getTopic().getV1Beta1Path());
```

Comment:
We are actually depending on the v1 API now. Is internal system still using v1beta1?